### ISSUE TITLE
Improvements to resources.cfg.in

### ISSUE THREAD

**Issue by akien-mga** [OpenDungeons/OpenDungeons#367]
Talking about the resources.cfg.in as of https://github.com/OpenDungeons/OpenDungeons/blob/2f2a9573377f589b6298525653ca44859d24af0a/cmake/config/resources.cfg.in

There might be some enhancements to do in this file, but as I don't really know OGRE's resource management, I open this issue as a RFC (request for comments).

What I see:
- `@OPENDUNGEONS_MEDIA_DIR@/` is useless as far as I can tell: we don't define it in CMake, and @Bertram25 implemented the "out of source" media dir in an other way. We should probably drop it then. If so, should we keep it as a file to configure in cmake/config?
- Due to the implementation of the data dir in the code, there doesn't seem to be an easy way to define absolute paths in this file, since all paths are prefixed with the actual data dir (e.g. `./` or `/usr/share/games/opendungeons/`). So if I want to add a system font dir as a resource, I need to use a path relative to the data dir, and that's a bit annoying to compute (since some distros might install OD in /usr/share/opendungeons and not /usr/share/games/opendungeons).
  Is there a way to achieve absolute paths in the resources.cfg syntax, or a change that could be made in the code to respect leading slashes as a mark of the root filesystem? Linked to #366.
- Is the file up to date? I see for example that there are three subdirectories linked in materials/RTShaderLib, but the repo contains more: GLSL150, GLSLES, HLSL.
- Do we want to keep materials/RTShaderLib/Cg? AFAIK we dropped the Cg plugin because it's nonfree.

**Comment by Bertram25** [comment]
1. Yes, IMO we should be able to drop @OPENDUNGEONS_MEDIA_DIR@/ and related stuff.
2. Completely duscussed in #366 Relative paths will have to be used for now. Either Ogre will move or we will have to keep using relative paths or add a custom resource loader.
3. & 4. I updated the RTShaderLib/ folder some time ago to the latest version. AFAIK, the Cg, GLSL150, GLSLES and HLSL folders can be dropped and the cg path removed from the plugin(_d).cfg path.

I'll make a PR and hopefully this won't break stuff for everyone.

**Comment by oyvindln** [comment]
Removing the HLSL folder will cause problems for people using Direct3D on windows. The GLSL150 folder may be used depending on OpenGL version, so I don't know if that should be removed. Cg is disabled, so we don't need that, and I would assume we don't have anyone using the game on an OpenGL ES platform so that is probably safe to remove too.

**Comment by Bertram25** [comment]
> Removing the HLSL folder will cause problems for people using Direct3D on windows.

Yes, but since we only propose the GL renderer at startup, it shouldn't be a problem.

> The GLSL150 folder may be used depending on OpenGL version, so I don't know if that should be removed.

That is my main fear. If Ogre auto select the GL shaders to be used between GLSL, GLSL150 and GLSLES. Then, IMHO, we shouldn't remove any of them. But if we're sure the GL renderer does use only one of the three, then it's ok. Do you how/where in its code Ogre select the shaders to use, by any chance?

**Comment by Bertram25** [comment]
Seen on windows when removing the Cg/ folder:

```
03:42:41: OGRE EXCEPTION(6:FileNotFoundException): Cannot locate resource FFPLib_Transform.cg in resource group Graphics or any other group. in ResourceGroupManager::openResource at ..\..\..\..\OgreMain\src\OgreResourceGroupManager.cpp (line 756)
03:42:41: High-level program 9bdf951e-8e41-6551-7864-f3cfa61453f7_VS encountered an error during loading and is thus not supported.
OGRE EXCEPTION(6:FileNotFoundException): Cannot locate resource FFPLib_Transform.cg in resource group Graphics or any other group. in ResourceGroupManager::openResource at ..\..\..\..\OgreMain\src\OgreResourceGroupManager.cpp (line 756)
03:42:41: OGRE EXCEPTION(2:InvalidParametersException): Could not create gpu programs from render state  in ProgramManager::acquireGpuPrograms at ..\..\..\..\..\Components\RTShaderSystem\src\OgreShaderProgramManager.cpp (line 108)
03:42:41: Internal Ogre3D exception: OGRE EXCEPTION(2:InvalidParametersException): Could not create gpu programs from render state  in ProgramManager::acquireGpuPrograms at ..\..\..\..\..\Components\RTShaderSystem\src\OgreShaderProgramManager.cpp (line 108)
```

I'll have to readd the Cg/ folder.

**Comment by Bertram25** [comment]
Woohoo. One last issue before release!
I still need to know what to do with direct X, though. I'll do the two maps in the meanwhile.

**Comment by akien-mga** [comment]
> I still need to know what to do with direct X, though. 

Ping @hwoarangmy, I know that he deactivated general email notifications :-)

**Comment by hwoarangmy** [comment]
Well, I don't use DirectX so I don't mind about removing it ^^
Anyway, as @akien-mga pointed, I think we should focus on one renderer and OpenGL seems better as it is cross platform...

**Comment by Bertram25** [comment]
Ok, I'd like to get @oyvindln 's point of view, too, as he was the one wanting us to keep it AFAIK.

**Comment by akien-mga** [comment]
Fixed by 85e649e, f25f18ed and 1884301.
